be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidatorSuite.scala

```diff
@@ -266,6 +270,48 @@ class ElectionValidatorSuite extends TestKit(ActorSystem("electionValidatorTestA
     system.actorOf(dbActorMock)
   }
 
+  private def mockDbForResultElection: AskableActorRef = {
+    val dbActorMock = Props(new Actor() {
+      override def receive: Receive = {
+        case DbActor.ReadLaoData(_) =>
+          sender() ! DbActor.DbActorReadLaoDataAck(laoDataForResultElection)
+        case DbActor.Catchup(_) =>
+          sender() ! DbActor.DbActorCatchupAck(List(MESSAGE_END_ELECTION_WORKING))
+        case DbActor.ReadSetupElectionMessage(_) =>
+          sender() ! DbActor.DbActorReadAck(Some(MESSAGE_SETUPELECTION_OPEN_BALLOT_WORKING))
+      }
+    })
+    system.actorOf(dbActorMock)
+  }
+
+  private def mockDbElectionNotEnded: AskableActorRef = {
+    val dbActorMock = Props(new Actor() {
+      override def receive: Receive = {
+        case DbActor.ReadLaoData(_) =>
+          sender() ! DbActor.DbActorReadLaoDataAck(laoDataForResultElection)
+        case DbActor.Catchup(_) =>
+          sender() ! DbActor.DbActorCatchupAck(List.empty)
+        case DbActor.ReadSetupElectionMessage(_) =>
+          sender() ! DbActor.DbActorReadAck(Some(MESSAGE_SETUPELECTION_OPEN_BALLOT_WORKING))
+      }
+    })
+    system.actorOf(dbActorMock)
+  }
+
+  private def mockDbWrongQuestionIds: AskableActorRef = {
+    val dbActorMock = Props(new Actor() {
+      override def receive: Receive = {
+        case DbActor.ReadLaoData(_) =>
+          sender() ! DbActor.DbActorReadLaoDataAck(laoDataForResultElection)
+        case DbActor.Catchup(_) =>
+          sender() ! DbActor.DbActorCatchupAck(List.empty)
```

Comment:
Shouldn't you also have the message of end election in your catchup list?
Maybe I am mistaken, but what you should do is to have the mockDb that works, but test it with a wrong message ResultElection. 
Since, what you try to test is if the ids of the ResultElection message match with the initial question ids.

Reply:
When checking the result election ids, I take the result election ids, and compare them with the ones present in the setupElection message, assuming that the setup election message, before being written in the DB has been validated and hence contains correct question ids. So no need for a catchup here, however, I added an end election message in the catchup for consistency.

Reply:
(Indeed this comment does not concern this line, should have put it lower ^^)

Reply:
Also, if you agree with my previous comments, you could totally remove the `mockDbWrongQuestionIds` and simply use the `mockDbForResultElection` for ResultElection messages with wrong Ids :).